src/Bundler/Runner/UnixSocket.php

```diff
@@ -18,25 +20,32 @@
  */
 class UnixSocket
 {
+    private $logger;
     private $socket;
 
-    public function __construct()
+    public function __construct(LoggerInterface $logger = null)
     {
+        $this->logger = $logger ?? new NullLogger();
         $this->socket = socket_create(AF_UNIX, SOCK_STREAM, 0);
+        socket_set_option($this->socket, SOL_SOCKET, SO_RCVTIMEO, ['sec' => 3, 'usec' => 0]);
     }
 
     public function connect(string $address): void
     {
-        if (! @socket_connect($this->socket, $address)) {
+        if (!@socket_connect($this->socket, $address)) {
+            $this->logger->debug('[UnixSocket] Connection exception');
             throw new SocketException('Problem connecting to the socket');
```

Comment:
What if we do `throw SocketException::connect($this->socket)` with `socket_strerror(socket_last_error())`?

We should probably catch the exception and _then_ log where we catch it. Right now we're logging _and_ throwing, which seems kinda cumbersome as it mainly logs and then throws. My suggestion is basically to push the logger dependency out of the unix socket and into the layers calling/catching this.

Same goes for all the other exceptions.